plugin/pkg/scheduler/algorithm/priorities/taint_toleration.go

```diff
@@ -27,56 +27,47 @@ import (
 
 // CountIntolerableTaintsPreferNoSchedule gives the count of intolerable taints of a pod with effect PreferNoSchedule
 func countIntolerableTaintsPreferNoSchedule(taints []v1.Taint, tolerations []v1.Toleration) (intolerableTaints int) {
-	for i := range taints {
-		taint := &taints[i]
+	for _, taint := range taints {
 		// check only on taints that have effect PreferNoSchedule
 		if taint.Effect != v1.TaintEffectPreferNoSchedule {
 			continue
 		}
 
-		if !v1.TolerationsTolerateTaint(tolerations, taint) {
+		if !v1.TolerationsTolerateTaint(tolerations, &taint) {
 			intolerableTaints++
 		}
 	}
 	return
 }
 
-// getAllTolerationEffectPreferNoSchedule gets the list of all Toleration with Effect PreferNoSchedule
+// getAllTolerationEffectPreferNoSchedule gets the list of all Tolerations with Effect PreferNoSchedule or with no effect.
 func getAllTolerationPreferNoSchedule(tolerations []v1.Toleration) (tolerationList []v1.Toleration) {
-	for i := range tolerations {
-		toleration := &tolerations[i]
+	for _, toleration := range tolerations {
 		if len(toleration.Effect) == 0 || toleration.Effect == v1.TaintEffectPreferNoSchedule {
```

Comment:
Makes sense, I added one more line at the top of function which tells that this function can collect all tolerations that have either no effect or PreferNoSchedule.

I added a test that has both empty toleration and taint. I will add one that has empty toleration and a taint separately.